src/test/java/dk/sdu/se_f22/contentmodule/stopwords/SemesterProjekt1.4_stopWordFilter/src/StopWords.java

```diff
@@ -0,0 +1,161 @@
+    public static final String DB_USER = "postgres";
+    protected Statement statement ;
+    protected Connection connection ;
+    private static Scanner input = new Scanner(System.in);
+    private static List<String> tokenTester = new ArrayList<>();
+
+   public StopWords() {}
+
+//Note: the method below is defined for testing purposes, and not a part of the CMS-4
+    public static List<String> unfiltered() {
+        if (tokenTester.isEmpty()) {
+            System.out.println("Type a token ");
+            String token = input.nextLine();
+            String[] split = token.split(" ", 0);
+
+            for (String s : split) {
+                 s = s.replaceAll("\\s+","-").replaceAll("'", "");
+                tokenTester.add(s);
+            }
+        }
+        return tokenTester;
+    }
+
+// open connect to the DB
+    public boolean open() {
+        try {
+            connection = DriverManager.getConnection(CONNECTION_STRING, DB_USER, DB_PASSWORD);
+            fromEngTokenPrep =connection.prepareStatement(WORDS_FROM_TOKEN_REMOVE);
+            insertWordsPrep = connection.prepareStatement(INSERT_DA_WORDS);
+            deleteWordsPrep =connection.prepareStatement(REMOVE_DA_WORDS);
+            queryWordsPrep =connection.prepareStatement(QUERY_DA_WORDS);
+            return true;
+
+        } catch (SQLException e) {
+            System.out.println("Couldn't connect to database : " + e.getMessage());
+            e.printStackTrace();
+            return false;
+        }
+    }
+    public void close() {
+        try {
+            if (connection != null) {
+                connection.close();
+            }
+        } catch (SQLException e) {
+            System.out.println("Couldn't close connection : " + e.getMessage());
+        }
+    }
+// filters a string token and removes words that match the list of stopWords in the DB
+    @Override
+    public Collection<String> filter(List<String> unfilteredTokens) {
+       open();
+        System.out.println(fromEngTokenPrep);
+               Collection<String> useableStopwords = new ArrayList<>();
+        try {
+            for (String i : unfilteredTokens) {
+                fromEngTokenPrep.setString(1,i);
+                ResultSet resultSet = fromEngTokenPrep.executeQuery();
+                while (resultSet.next()) {
+                    useableStopwords.add(resultSet.getString("name"));
+                }
+                resultSet.close();
+            }
+            unfilteredTokens.removeAll(useableStopwords);
+            fromEngTokenPrep.close();
+            connection.close();
+
+
+        } catch (SQLException e) {
+            System.out.println("Error connecting to database" + e.getMessage());
+            e.printStackTrace();
+        }
+        return unfilteredTokens;
+
+    }
+
+// Adding new stopWords to the list in the DB
+    @Override
+    public void addStopWord(String stopWord) {
+       open();
+        try {

```

Comment:
You could utilize `try-with-resources` here, which means you don't have to make sure it's open, nor make sure to close the resources.

something like:
```java
try (Connection conn = DBConnection.getPooledConnection();
       PreparedStatement pStmt = conn.prepareStatement("sql query");
) {
// No need for closing the connection nor the prepared statement since it's automatically done.
} catch () {}